.github/summarize_iterations.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+from datetime import datetime
+from datetime import date
+from collections import defaultdict
+import csv
+from datetime import date
+import pandas as pd
+import json
+
+
+def summarize_iterations():
+    """Divide the issues csv by iteration and by whether it contains the `design` label.
+    For dev and for design, summarize the story points and issues closed during 
+    the each iteration. Save the output json to the `data` dir as iteration-summary.json.
+    """
+    df = pd.read_csv('data/issues.csv')
+    df['closed'] = [pd.to_datetime(x.split('T')[0]) if not pd.isna(x) else None for x in df['closed']]
+
+    df_closed = df[~pd.isna(df['closed'])]
+    mentions_design = df_closed['labels'].apply(lambda x: ('design' in x.split(';')) if not pd.isna(x) else False)
+    df_design = df_closed[mentions_design]
+    df_dev = df_closed[~mentions_design]
+
+    with open('data/iterations.json') as f:
+        iterations_json = json.load(f)
+
+    # Build and save the summary CSV 
+    output_l = []
+    for iteration_d in iterations_json:
+        row = {}
+        
+        start_date = datetime.fromisoformat(iteration_d['from'])
```

Comment:
It looks like this is recalculating all the iterations. Can we keep track of when it was last run and only recalculate any iterations since that date?

My concern is that opening and closing tickets across iterations (rare, but possible) would result in an iteration total being changed after the fact, which isn't the behavior I want.

Reply:
We are only appending rows to issues.csv, so an issue would show up twice with a different closed date. The only way an iteration would be changed after the fact is if you changed the dates in iterations.json